src/Products/PageTemplates/tests/test_ptfile.py

```diff
@@ -15,15 +15,17 @@
 
 class TypeSniffingTestCase(unittest.TestCase):
 
-    TEMPFILENAME = tempfile.mktemp(".zpt")
-
     def setUp(self):
         from Products.PageTemplates.interfaces import \
             IUnicodeEncodingConflictResolver
         from Products.PageTemplates.unicodeconflictresolver import \
             DefaultUnicodeEncodingConflictResolver
         from zope.component import provideUtility
 
+        self.tmpfile = tempfile.NamedTemporaryFile(suffix=".zpt", delete=False)
+        self.TEMPFILENAME = self.tmpfile.name
+        self.tmpfile.close()
```

Comment:
We can use [`NamedTemporaryFile`](https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile) as a context manager, so the last line is done automatically

Reply:
I changed those places to use a context manager